lib/web_catalog/api_extractor.es6.js

```diff
@@ -0,0 +1,406 @@
+            } else {
+              // This is not a object library, try to add its property to
+              // it's prototype. For example, document.body.style contains
+              // meaningful properties of CSSStyleDeclaration, then its
+              // properties will be added to CSSStyleDeclaration interface.
+              if (!className || className === 'Object') return;
+              // Skip if its class is Object, or cannot find it's class name.
+              // We don't want to add extra properties for Object.
+              if (!map[className]) map[className] = [];
+              this.arrayMerge_(map[className],
+                this.getObjectProperties_(og, id));
+            }
+          }
+          if (og.getPrototype(id)) {
+            // Visit object __proto__ if its __proto__ exists.
+            this.extractAPI_(og, map, og.getPrototype(id),
+              `${name}.__proto__`, visited, {proto: true});
+          }
+        }
+
+        // Don't visit __proto__'s properties
+        // since __proto__ does not contain instances.
+        if (proto) return;
+
+        // Visit object's properties.
+        let propertiesIdMap = og.getPropertiesIds(id);
+        let keys = Object.keys(propertiesIdMap);
+        for (let i = 0; i < keys.length; i++) {
+          let cleanUpKey = this.cleanUp_(keys[i]);
+          if (this.blacklistProperties.indexOf(cleanUpKey) === -1 &&
+            cleanUpKey !== 'prototype') {
+            // Visit object if its is not blacklisted or prototype.
+            this.extractAPI_(og, map, propertiesIdMap[keys[i]],
+              `${name}.${cleanUpKey}`, visited);
+          }
+        }
+      },
+    },
+    {
+      name: 'extractWebCatalog',
+      description: 'This function reads an object graph and produce' +
+        'a web catalog JSON',
+      code: function(og, opt) {
+        opt = opt || {};
+        this.defaults.retainBuildInMembers = opt.retainBuildInMembers ||
+          this.defaults.retainBuildInMembers;
+        this.defaults.retainConstantMembers = opt.retainConstantMembers ||
+          this.defaults.retainConstantMembers;
+        const apiCatalogs = {};
+        // Read root ID, should be the id of window object.
+        this.metadata.rootId = og.getRoot();
+        // visited is a map keep track of visited objects.
+        const visited = {rootId: true};
+        this.getFunctionBuildInProperties_(og);
+        this.getObjectBuildInProperties_(og);
+        // Get all window.X as first level objects.
+        const firstLevelObjects = og.getObjectKeys(this.metadata.rootId);
+        for (let i = 0; i < firstLevelObjects.length; i++) {
+          let interfaceObject = this.cleanUp_(firstLevelObjects[i]);
+          let objectId = og.lookup(interfaceObject, this.metadata.rootId);
+          this.extractAPI_(og, apiCatalogs, objectId, interfaceObject,
+            visited, {firstLevel: true});
+        }
+        // Window interface is a special case,
```

Comment:
I will remove this loop and window's APIs in previous loop. 
In previous strategy, not all objects under window object are considered as window's APIs. This is way a second loop was used.